make it possible to copy cfg correctly

Kyle-Kyle opened the pull request:
This commit is aiming at resolving issue #1316 .

I know it may be not right to copy attributes in such a lazy way, but I could not find a reason not to.
So if you think there is something wrong with this implementation, please tell me. And I will fix it ASAP.

ltfish commented:
Can you please add a test case to it? You probably want to skip all attributes wrapped inside `__xxx__`, too. Other than that, I totally don't care about the hackishness of this solution :)

ltfish commented on tests/test_cfgfast.py:
```diff
@@ -417,6 +417,24 @@ def test_segment_list_6():
+#
+
+def test_cfg_copy():
+    path = os.path.join(test_location, "cgc/CADET_00002")
```
`os.path.join(test_location, "cgc", "CADET_00002")`

ltfish commented on tests/test_cfgfast.py:
```diff
@@ -417,6 +417,24 @@ def test_segment_list_6():
+
+    cfg = proj.analyses.CFGFast()
+    cfg_copy = cfg.copy()
+    for attr, value in cfg_copy.__dict__.items():
```
`value` is unused, and pylint complains about it ;)

ltfish commented:
Looks good to me! Fix linting and I'll hit merge. Thank you!

ltfish commented:
You have a failed test case :)

```
FAIL: test_cfgfast.test_tail_call_optimization_detection_armel
```

Kyle-Kyle commented:
The weird thing is: I rebased on master and ran the test for multiple times. It worked without error.

And I modified only `copy` functions which are not used by angr internally(to my best knowledge). So in theory, it should not affect other functionality.

Can we just rerun CI again?

ltfish commented:
Just did! It might be a glitch.

Kyle-Kyle commented:
still? I will have look into it. But I can't replicate this behavior on my own machine.

Kyle-Kyle commented:
I checked the error message, the `set` `all_func_addrs` on travis is a true subset of `all_func_addrs` on my machine, which leads to the error.
Still no idea why

rhelmot commented:
that failure also showed up on [my branch](https://travis-ci.org/angr/angr/jobs/481942059#L1670) so I think it might also be happening on master. I'm going to merge mine and I think this should be merged too.

ltfish commented:
I'll hit merge and then figure out what's going on with that test case.

ltfish commented:
Merged. Thank you!
